backend/app/controllers/spree/admin/resource_controller.rb

```diff
@@ -134,9 +134,14 @@ def belongs_to(model_name, options = {})
     end
   end
 
+  def resource_error_filter(options = {})
+    options[:flash_class] ||= "Spree::#{parent_model_name.classify}".constantize
+    options[:redirect_url] ||= spree.new_polymorphic_url([:admin, parent_model_name])
+    base_resource_not_found(options)
+  end
```

Comment:
I think if we capture the exceptions specifically in the methods that would raise them, we can make this method unnecessary.